api-flask/app/tests/test_api.py

```diff
@@ -113,6 +152,29 @@ def test_stats_endpoint2():
     assert response.status_code == 200
 
 
+def test_stats_auth():
```

Comment:
To test fully it would be interesting to have another tests that returns an unauthorized error